### DISCUSSION TITLE
GH discussions [meta post]

### DISCUSSION THREAD

**Discussion by Daeraxa** [pulsar-edit/.github#31] [General]
Conversation on Discord around [here](https://discord.com/channels/992103415163396136/992275411713134623/1024353233864511488) onwards.

I think it is a bit of a no-brainer, especially at the moment, to use GH discussions for our central "community" area for longer form topics.

The alternatives would be:
- Discourse (or similar)
    - Can get a free plan for OSS project (https://free.discourse.group/)
- Reddit
    - We already have this set up but its not ideal for this purpose
- Discord "forums"
    - Already set up but really not good for longer term referencing etc.

So assuming GH discussions are the way forwards I think the following needs to be done:
- Archive or move discussions from the repo discussions to the org discussion
- Close repo discussions
- Org discussion currently tied to `pulsar-edit.github.io`, should this be moved to `.github` (or another if not possible)?
- Change categories to make sense and use labels as a "tag" system for subcategories.
e.g.

```
- Announcements
- Pulsar
- Website
- Documentation
- Polls

-Labels: Idea, Question, Help, Feedback, Core, ppm, Server, Release, Beta, API, Backend, Website, Docs
```

We would obviously need to add the link to our socials lists.

**Comment by confused-Techie** [comment]
This seems like a great idea to me. My only concerns would be replicating the data that already current exists in discussions. Additionally moving to `.github` would make sense as its the org repo. Although we could do `pulsar` as our name sake repo. But really it doesn't matter all that much, so either one I'm fine with. 

As for moving to something other than Discord or GitHub I'd personally vote a hard no. Personally its already some effort to keep up with communication on two fronts. So maybe we could obviously keep Discord for current communication, then instead of polls on there we move those over to GitHub? Since those usually contain the longer form conversations that would be needed to reference back to.

Otherwise if we wanted to get more long form content on GitHub discussions that seems difficult. Since I feel that many of our long form stuff comes from small chat. Which we would never then know what constitutes its own big Discussion post and what doesn't. And if everything is a Discussion post we are back at square one with being able to filter all of it.

So maybe keep discussion mostly on Discord, polls on GitHub Discussions, then IF needed, an important conversation that came from Discord and caused a big change can have more an announcement post on GitHub discussions. Basically heres the change we are making and here is why, referencing to what some people said and what not. 

Because as much as we try, structing how we communicate won't be easy, and we don't want it to be bothersome.

So in short, fully on board with keeping only one place for discussions. Maybe moving polls to these discussions. Then changes that have arisen out of normal chat on Discord can get there own post on the Discussions.

**Comment by ghost** [comment]
This seems like a bad idea. I don't like having our discussions really being pushed. Here is why I don't like it (to make it clear):

- **Labels:** Labels seem good, but they aren't good enough. For example, you have to create labels that can flow through issues and pull requests, not strictly for discussions. Categories are strictly for discussions.
- **Moving discussions to `.github`:** I'm actually fine doing this, as long as we transfer all posts from `pulsar-edit.github.io` to our proposed repository.
- ** @confused-Techie 's comment about Discord being main chat:** I honestly, **DO NOT LIKE THAT AT ALL.** GitHub Discussions is really what I like to use. I sometimes use Discord, only when I feel like I need to.

That's all I have to say.

**Reply by Daeraxa** [reply]
I'm not sure I understand all of your points? 

> This seems like a bad idea. I don't like having our discussions really being pushed 
> GitHub Discussions is really what I like to use

But this is exactly what @confused-Techie and I are both advocating, making a central area for discussion on GH. Or am I not understanding what you mean here?

As for the labels, I think it is a very common thing and is clearly supported here as you can add labels to a discussion. For example if I made a post in `Pulsar` I might want to highlight it as a `Help` or `Question`. Not to mention assistance in categorising announcement posts and the ability to use those labels at a later date to filter through search results.

![image](https://user-images.githubusercontent.com/58074586/192668275-1a1aa4a7-ba74-4478-9f2f-a648b63dd17e.png)

I don't think non-members should be allowed to assign labels to issues or PRs anyway, that should be the job of the triage team, GH actions should be assigning some basic ones based on the type of issue ideally too.

What kind of problems do you see by using labels here and in the PR and issue workflows? (not being facetious, genuinely wondering).

>  about Discord being main chat:** I honestly, DO NOT LIKE THAT AT ALL. 

Unfortunately, at least at present, you are very much in the minority, Discord is where everything is happening at the moment. Right now there are about 6 or 7 people all talking about 4 or 5 different topics in a way that I just don't think is possible on a "forum" type discussion.

Don't get me wrong, I do think we should be using GH discussions far more than we are which is why I've started posting stuff, why I made this topic and why I'm advocating that we make it clear how we want to use them (if people agree) so we can get stuff set up and ready.

**Comment by kaosine** [comment]
I mean I could see trying discourse since its stuff is better setup imo. But the moving to as much as possible being in the root org discussion would probably be ideal. Heck, I almost wonder if somehow there could be a way to properly integrate a discord alternative, so we can reply to things outside GitHub or have it post there and sync up. Searching for this kind of thing is a pain because mentioning GitHub in a search want to pull up people's repos just mentioning integration or dashboard anywhere in their repo XD
